pym/bob/cmds/build.py

```diff
@@ -848,7 +848,7 @@ def _cookPackageStep(self, packageStep, checkoutOnly, depth):
             # downloads are not possible anymore. Even if the package was
             # previously downloaded the oldInputHashes will be None to trigger
             # an actual build.
-            if not wasDownloaded:
+            if (not wasDownloaded) or self.__force:
```

Comment:
This will also cause re-builds of packages that just been downloaded, e.g. when building in an empty workspace with `-f`. I think the solution is to add `or self.__force` at line 822. This will wipe out the result and either download or build the package.